### PR TITLE
Fix position when no serial positions

### PR DESCRIPTION
**Pull request by jpalumickas**
Opened branch from https://github.com/swanandp/acts_as_list/pull/208 to fix tests and rebase to master branch.

Can you take a look @brendon ?

### CODE REVIEW

**Comment by jpalumickas** [comment]
@brendon There is one test failing, I think it's random fail, can you restart a build for this pull request ?

**Comment by jpalumickas** [comment]
I restarted a build with amend, now it's passing. It was a random fail.

File: lib/acts_as_list/active_record/acts/list.rb [hunk 1]
```diff
@@ -174,8 +174,12 @@ def move_lower
           return unless lower_item
 
           acts_as_list_class.transaction do
-            lower_item.decrement_position
-            increment_position
+            if lower_item.send(position_column) != self.send(position_column)
```
Comment (brendon) [inline lib/acts_as_list/active_record/acts/list.rb]: Just wondering with these methods. Do we need the branching here. Could the first branch not handle all situations (i.e. consecutive numbering and non-consecutive?)

File: test/shared_list_sub.rb [hunk 2]
```diff
@@ -53,9 +79,9 @@ def test_next_prev_groups
       assert_equal [li2, li3], li1.lower_items(2)
       assert_equal [], li4.lower_items
 
-      assert_equal [li1, li2], li3.higher_items
+      assert_equal [li2, li1], li3.higher_items
```
Comment (brendon) [inline test/shared_list_sub.rb]: Is it going to break existing code if the order of returned higher items is in reverse to what it used to be?

**Comment by brendon** [comment]
Thanks @jpalumickas, I just had a couple of queries. If we can settle those then I think this can be merged :) Closes #208 (replaced by this PR).

File: lib/acts_as_list/active_record/acts/list.rb (hunk 1 above)
Comment (jpalumickas) [inline lib/acts_as_list/active_record/acts/list.rb]: @PoslinskiNet can you answer that ? 😉

Comment (PoslinskiNet) [inline lib/acts_as_list/active_record/acts/list.rb]: Yes, it is possible, but didn't want to change default behaviour, just handle "non-consequence" branch, so consider that as a refactor :).

Comment (brendon) [inline lib/acts_as_list/active_record/acts/list.rb]: Ok, yes that makes sense.

**Comment by brendon** [comment]
Just a note. I have another possibly compatibility-breaking change here: https://github.com/swanandp/acts_as_list/pull/220 so it might make sense to bump the minor version to 0.8.0 to take both of these two changes into account. At least people will then double-think blindly upgrading. We should probably document these changes in the README with an upgrade message.

**Comment by swanandp** [comment]
Agree on both counts, version bump and doc updates

**Comment by brendon** [comment]
0.8.0 has been released and includes this change.

**Comment by jpalumickas** [comment]
Perfect, thanks 👍

**Comment by PoslinskiNet** [comment]
Nice 👍

**Pull request merged**